downstream/modules/eda/proc-eda-set-up-credential-types.adoc

```diff
@@ -0,0 +1,169 @@
+You can create a credential type to use with a source plugin or a private container registry that you select based on the supported, default credential types. You can make your credential type available to a team or individuals.
+
+
+.Procedure
+
+. Log in to the {PlatformNameShort} Dashboard.
+. From the navigation panel, select {MenuADCredentialType}.
+. Click btn:[Create credential type].
+. Insert the following:
++
+Name:: Insert the name.
+Description:: This field is optional.
+. In the *Input Configuration* field, specify an input schema that defines a set of ordered fields for that type. The format can be in YAML or JSON:
++
+*YAML*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+fields:
+  - type: string
+    id: username
+    label: Username
+  - type: string
+    id: password
+    label: Password
+    secret: true
+required:
+  - username
+  - password
+----
++
+
+View more YAML examples at the link:https://yaml.org/spec/1.2.2/[YAML page].
++
+*JSON*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+"fields": [
+  {
+  "type": "string",
+  "id": "username",
+  "label": "Username"
+  },
+  {
+  "secret": true,
+  "type": "string",
+  "id": "password",
+  "label": "Password"
+   }
+  ],
+ "required": ["username", "password"]
+}
+----
++
+View more JSON examples at link:https://www.json.org/json-en.html[The JSON website].
+
+
+
+. In the *Injector Configuration* field, enter environment variables or extra variables that specify the values a credential type can inject.
+The format can be in YAML or JSON (see examples in the previous step).
++
+The following configuration in JSON format shows each field and how they are used:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+
+{
+  "file": {
+      "template": "[mycloud]\ntoken={{ api_token }}"
+  },
+  "env": {
+      "THIRD_PARTY_CLOUD_API_TOKEN": "{{ api_token }}"
+  },
+  "extra_vars": {
+      "some_extra_var": "{{ username }}:{{ password }}"
+  }
+}
+----
++
+Credential Types can also generate temporary files to support `.ini` files or certificate or key data:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "file": {
+      "template": "[mycloud]\ntoken={{ api_token }}"
+  },
+  "env": {
+      "MY_CLOUD_INI_FILE": "{{ tower.filename }}"
+  }
+}
+----
++
+In this example, {EDAcontroller} writes a temporary file that has:
++
+[literal, options="nowrap" subs="+attributes"]
+----
+[mycloud]\ntoken=SOME_TOKEN_VALUE
+----
++
+The absolute file path to the generated file is stored in an environment variable named `MY_CLOUD_INI_FILE`.
++
+The following is an example of referencing many files in a custom credential template:
++
+*Inputs*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "fields": [{
+    "id": "cert",
+    "label": "Certificate",
+    "type": "string"
+  },{
+    "id": "key",
+    "label": "Key",
+    "type": "string"
+  }]
+}
+----
++
+*Injectors*
++
+[literal, options="nowrap" subs="+attributes"]
+----
+{
+  "file": {
```

Comment:
@jself-sudoku file and env are not supported yet

Reply:
Thanks. Updated to reflect a similar code as what's used in a previous injector sample. Hope it's okay to use this.

Reply:
@mkanoor - I didn't know what example to use here since file and env were the only things included. I ended up using the same example I used above in step 6, but I don't think it's right. Do you have an example that would be right for this section?